gdal/port/cpl_http.cpp

```diff
@@ -940,14 +969,25 @@ CPLHTTPResult *CPLHTTPFetchEx( const char *pszURL, CSLConstList papszOptions,
         {
             const char* pszFormFileName = CSLFetchNameValue( papszOptions,
                 "FORM_FILE_NAME" );
+            const char* pszFilename = CPLGetFilename( pszFormFilePath );
             if( pszFormFileName == nullptr )
             {
-                pszFormFileName = CPLGetFilename( pszFormFilePath );
+                pszFormFileName = pszFilename;
             }
 
 #if LIBCURL_VERSION_NUM >= 0x073800 /* 7.56.0 */
-            curl_mime_name(mimepart, pszFormFileName);
-            curl_mime_filedata(mimepart, pszFormFilePath);
+            VSIStatBufL sStat;
+            if( VSIStatL( pszFormFilePath, &sStat ) == 0)
+            {
+                VSILFILE *mime_fp = VSIFOpenL( pszFormFilePath, "rb" );
```

Comment:
is it intended that no error is emitted when then stat() or open() fail ?

Reply:
There is no problem to send form without file. But it seems to me that check file exists is reasonable. I'll add checks for this case.